src/backends/sqlite3/statement.cpp

```diff
@@ -182,6 +182,9 @@ sqlite3_statement_backend::load_rowset(int totalRows)
                             col.buffer_.data_ = (col.buffer_.size_ > 0 ? new char[col.buffer_.size_] : NULL);
                             memcpy(col.buffer_.data_, sqlite3_column_blob(stmt_, c), col.buffer_.size_);
                             break;
+
+                        case dt_xml:
+                            throw soci_error("XML data type is not supported");
```

Comment:
Please provide a comprehensive handling of these new types for SQLite3 in a separate PR. The above lines are just placeholders that make the switch coverage complete for the enumeration type that is not SQLite3-specific.